cmd/crates/soroban-spec-typescript/fixtures/test_custom_types/src/index.ts

```diff
@@ -231,392 +231,807 @@ function ComplexEnumFromXdr(base64Xdr: string): ComplexEnum {
 const Errors = [ 
 {message:"Unknown error has occured"}
 ]
-export async function hello({hello}: {hello: string}, {signAndSend, fee}: {signAndSend?: boolean, fee?: number} = {signAndSend: false, fee: 100}): Promise<string> {
-    let invokeArgs: InvokeArgs = {
-        signAndSend,
-        fee,
-        method: 'hello', 
-        args: [((i) => xdr.ScVal.scvSymbol(i))(hello)], 
-    };
-    
-    // @ts-ignore Type does exist
-    const response = await invoke(invokeArgs);
-    return scValStrToJs(response.xdr) as string;
-}
-
-export async function woid( {signAndSend, fee}: {signAndSend?: boolean, fee?: number} = {signAndSend: false, fee: 100}): Promise<void> {
-    let invokeArgs: InvokeArgs = {
-        signAndSend,
-        fee,
-        method: 'woid', 
-        
-    };
-    
-    // @ts-ignore Type does exist
-    const response = await invoke(invokeArgs);
-    return ;
-}
-
-export async function val( {signAndSend, fee}: {signAndSend?: boolean, fee?: number} = {signAndSend: false, fee: 100}): Promise<any> {
-    let invokeArgs: InvokeArgs = {
-        signAndSend,
-        fee,
-        method: 'val', 
-        
-    };
-    
-    // @ts-ignore Type does exist
-    const response = await invoke(invokeArgs);
-    return scValStrToJs(response.xdr) as any;
-}
-
-export async function u32_fail_on_even({u32_}: {u32_: u32}, {signAndSend, fee}: {signAndSend?: boolean, fee?: number} = {signAndSend: false, fee: 100}): Promise<Result<u32>> {
-    let invokeArgs: InvokeArgs = {
-        signAndSend,
-        fee,
-        method: 'u32_fail_on_even', 
-        args: [((i) => xdr.ScVal.scvU32(i))(u32_)], 
-    };
-    
-    try {
-        
-    // @ts-ignore Type does exist
-    const response = await invoke(invokeArgs);
-    return new Ok(scValStrToJs(response.xdr) as u32);
-    } catch (e) {
-        //@ts-ignore
-        let err = getError(e.message);
-        if (err) {
-            return err;
-        } else {
-            throw e;
-        }
-    }
+export async function hello<R extends ResponseTypes = undefined>({hello}: {hello: string}, options: {
```

Comment:
would be super cool to run e.g. `prettier` on this code so that it's easier for consumers to grok and read docs inline (plus would help clean up diffs etc.), but I'm not sure how annoying that would be to add to the pipeline

Reply:
Yeah we definitely considered it but wanted to keep it simple for the first iteration. Perhaps with https://github.com/rome/tools